torch/utils/_device.py

```diff
@@ -0,0 +1,62 @@
+import torch
+from torch.overrides import TorchFunctionMode
+import functools
+
+@functools.lru_cache(1)
+def _device_constructors():
+    return {
```

Comment:
Feels like this can easily get out of date? :/

Reply:
The OpInfo test hopefully will help; in particular, we will notice if you failed to match a constructor but it is exercised in OpInfo. But yes, the implementation strategy here requires us to know which set of Python binding functions we need to match against. For extra safety, maybe we could have codegen dump out a list of schemas that have device arguments, but you're still on the hook for making sure that other factory functions implemented purely in Python are torch function'able. Which, BTW, we also have a test for in test_overrides.py.